### ISSUE TITLE
Support for Asynchronous Circuits

### ISSUE THREAD

**Issue by sirinath** [ucb-bar/chisel#206]
Is it possible to add support for building asynchronous circuits and also provide some tutorial on how this can be done on Chisel.

**Comment by da-steve101** [comment]
My approach to this might be class Analog extends Node
Have it as a double value that you can poke every clock for the emulator
Verilog.scala shouldn't need much modification
Could eventually have a PCB backend too and generate schematics and layouts
Would need Analog to have various properties (capacitance/inductance between etc) ... would be alot of work but doable. Thoughts?

**Comment by sirinath** [comment]
I am not sure how this can be modeled internally but would be a good addition. Also if Chisel tool chain becomes available on Windows I can start using these features.

**Comment by schoeberl** [comment]
Are you mixing here two topics: asynchronous and analog circuits?

Asynchronous circuits are still digital circuits, but with local handshaking instead of a clock.

Martin

**Comment by sirinath** [comment]
No. Asynchronous logic. https://en.wikipedia.org/wiki/Asynchronous_circuit

No global clock or there is only a local clock. https://en.wikipedia.org/wiki/Clock_gating

**Comment by schoeberl** [comment]
Why does then the original post contain the words “Analog class” and “double value poking
at every clock”?

Martin

**Comment by da-steve101** [comment]
Sorry yes, had something else on my mind when i was writing it. Didn't read the question

**Comment by da-steve101** [comment]
I guess for async could just allow a register to have a null clock
